### PR TITLE
[Fix](agg) fix bitmap agg core dump when phmap pointer assert alignment

### LINKED ISSUE #13380: [Bug] BE will be core dump phmap Pointer assert alignment
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

`#0 __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1 0x00007f565a863859 in __GI_abort () at abort.c:79
#2 0x00007f565a863729 in __assert_fail_base (fmt=0x7f565a9f9588 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n",
assertion=0x563628fbc2e0 "reinterpret_cast<uintptr_t>(p) % alignment == 0",
file=0x563628fbc260 "/home/zcp/repo_center/doris_master/thirdparty/installed/include/parallel_hashmap/phmap_base.h", line=3967,
function=<optimized out>) at assert.c:92
#3 0x00007f565a875006 in _GI__assert_fail (assertion=0x563628fbc2e0 "reinterpret_cast<uintptr_t>(p) % alignment == 0",
file=0x563628fbc260 "/home/zcp/repo_center/doris_master/thirdparty/installed/include/parallel_hashmap/phmap_base.h", line=3967,
function=0x563628fbbc40 "phmap::priv::internal_layout::CopyConst<Char, phmap::priv::internal_layout::LayoutImpl<std::tuple<_Tps ...>, phmap::integer_sequence<long unsigned int, SizeSeq ...>, phmap::integer_sequence<long unsig"...) at assert.c:101
#4 0x000056362ce4337f in phmap::priv::internal_layout::LayoutImpl<std::tuple<phmap::priv::btree_node<phmap::priv::map_params<unsigned int, roaring::Roaring, phmap::Less<unsigned int>, std::allocator<std::pair<unsigned int const, roaring::Roaring> >, 256, false> >, unsigned char, phmap::priv::map_slot_type<unsigned int, roaring::Roaring>, phmap::priv::btree_node<phmap::priv::map_params<unsigned int, roaring::Roaring, phmap::Less<unsigned int>, std::allocator<std::pair<unsigned int const, roaring::Roaring> >, 256, false> >>, phmap::integer_sequence<unsigned long, 0ul, 1ul, 2ul, 3ul>, phmap::integer_sequence<unsigned long, 0ul, 1ul, 2ul, 3ul> >::Pointer<1ul, char const> (this=0x7f39aed82da0, p=0xbebebebebebebebe <error: Cannot access memory at address 0xbebebebebebebebe>)
at /home/zcp/repo_center/doris_master/thirdparty/installed/include/parallel_hashmap/phmap_base.h:3967
#5 0x000056362ce353d5 in phmap::priv::btree_node<phmap::priv::map_params<unsigned int, roaring::Roaring, phmap::Less<unsigned int>, std::allocator<std::pair<unsigned int const, roaring::Roaring> >, 256, false> >::GetField<1l> (this=0xbebebebebebebebe)
at /home/zcp/repo_center/doris_master/thirdparty/installed/include/parallel_hashmap/btree.h:1162
#6 0x000056362ce2593e in phmap::priv::btree_node<phmap::priv::map_params<unsigned int, roaring::Roaring, phmap::Less<unsigned int>, std::allocator<std::pair<unsigned int const, roaring::Roaring> >, 256, false> >::leaf (this=0xbebebebebebebebe)
at /home/zcp/repo_center/doris_master/thirdparty/installed/include/parallel_hashmap/btree.h:1178
#7 0x000056362ce241c0 in phmap::priv::btree<phmap::priv::map_params<unsigned int, roaring::Roaring, phmap::Less<unsigned int>, std::allocator<std::pair<unsigned int const, roaring::Roaring> >, 256, false> >::internal_clear (this=0x625025998018,
node=0xbebebebebebebebe) at /home/zcp/repo_center/doris_master/thirdparty/installed/include/parallel_hashmap/btree.h:3218
#8 0x000056362ce1636e in phmap::priv::btree<phmap::priv::map_params<unsigned int, roaring::Roaring, phmap::Less<unsigned int>, std::allocator<std::pair<unsigned int const, roaring::Roaring> >, 256, false> >::clear (this=0x625025998018)
at /home/zcp/repo_center/doris_master/thirdparty/installed/include/parallel_hashmap/btree.h:2855
#9 0x000056362ce1380c in phmap::priv::btree<phmap::priv::map_params<unsigned int, roaring::Roaring, phmap::Less<unsigned int>, std::allocator<std::pair<unsigned int const, roaring::Roaring> >, 256, false> >::~btree (this=0x625025998018, __in_chrg=<optimized out>)
at /home/zcp/repo_center/doris_master/thirdparty/installed/include/parallel_hashmap/btree.h:1698
#10 0x000056362cdfd1ec in phmap::priv::btree_container<phmap::priv::btree<phmap::priv::map_params<unsigned int, roaring::Roaring, phmap::Less<unsigned int>, std::allocator<std::pair<unsigned int const, roaring::Roaring> >, 256, false> > >::~btree_container (
this=0x625025998018, __in_chrg=<optimized out>)
at /home/zcp/repo_center/doris_master/thirdparty/installed/include/parallel_hashmap/btree.h:3260
#11 0x000056362cdfd208 in phmap::priv::btree_set_container<phmap::priv::btree<phmap::priv::map_params<unsigned int, roaring::Roaring, phmap::Less<unsigned int>, std::allocator<std::pair<unsigned int const, roaring::Roaring> >, 256, false> > >::~btree_set_container (
this=0x625025998018, __in_chrg=<optimized out>)
at /home/zcp/repo_center/doris_master/thirdparty/installed/include/parallel_hashmap/btree.h:3415
#12 0x000056362cdfd224 in phmap::priv::btree_map_container<phmap::priv::btree<phmap::priv::map_params<unsigned int, roaring::Roaring, phmap::Less<unsigned int>, std::allocator<std::pair<unsigned int const, roaring::Roaring> >, 256, false> > >::~btree_map_container (
this=0x625025998018, __in_chrg=<optimized out>)
at /home/zcp/repo_center/doris_master/thirdparty/installed/include/parallel_hashmap/btree.h:3573
#13 0x000056362cdfd240 in phmap::btree_map<unsigned int, roaring::Roaring, phmap::Less<unsigned int>, std::allocator<std::pair<unsigned int const, roaring::Roaring> > >::~btree_map (this=0x625025998018, __in_chrg=<optimized out>)
at /home/zcp/repo_center/doris_master/thirdparty/installed/include/parallel_hashmap/btree.h:3930
#14 0x000056362cdfd384 in doris::detail::Roaring64Map::~Roaring64Map (this=0x625025998018, __in_chrg=<optimized out>)
at /home/zcp/repo_center/doris_master/be/src/util/bitmap_value.h:95
#15 0x000056362ce07018 in doris::BitmapValue::~BitmapValue (this=0x625025998010, __in_chrg=<optimized out>)
at /home/zcp/repo_center/doris_master/be/src/util/bitmap_value.h:1141
#16 0x000056362fe11cee in doris::vectorized::AggregateFunctionBitmapData<doris::vectorized::AggregateFunctionBitmapIntersectOp>::~AggregateFunctionBitmapData (this=0x625025998010, __in_chrg=<optimized out>)
at /home/zcp/repo_center/doris_master/be/src/vec/aggregate_functions/aggregate_function_bitmap.h:88
#17 0x000056362fe11d16 in doris::vectorized::IAggregateFunctionDataHelper<doris::vectorized::AggregateFunctionBitmapData<doris::vectorized::AggregateFunctionBitmapIntersectOp>, doris::vectorized::AggregateFunctionBitmapOp<doris::vectorized::AggregateFunctionBitmapIntersectOp> >::destroy (this=0x60700161b1a0, place=0x625025998010 '\276' <repeats 200 times>...)
at /home/zcp/repo_center/doris_master/be/src/vec/aggregate_functions/aggregate_function.h:389
#18 0x00005636308b259b in doris::vectorized::AggregationNode::_destroy_agg_status (this=0x61e00107a080,
data=0x625025998010 '\276' <repeats 200 times>...) at /home/zcp/repo_center/doris_master/be/src/vec/exec/vaggregation_node.cpp:509
#19 0x00005636308b6b5b in doris::vectorized::AggregationNode::_close_without_key (this=0x61e00107a080)
at /home/zcp/repo_center/doris_master/be/src/vec/exec/vaggregation_node.cpp:668
#20 0x0000563630a63ead in std::__invoke_impl<void, void (doris::vectorized::AggregationNode::&)(), doris::vectorized::AggregationNode&> (
__f=@0x603017f3e9c0: (void (doris::vectorized::AggregationNode::*)(class doris::vectorized::AggregationNode * const)) 0x5636308b69fa <doris::vectorized::AggregationNode::_close_without_key()>, __t=@0x603017f3e9d0: 0x61e00107a080)
at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:74
#21 0x0000563630a5886f in std::__invoke_r<void, void (doris::vectorized::AggregationNode::&)(), doris::vectorized::AggregationNode&>
(
__fn=@0x603017f3e9c0: (void (doris::vectorized::AggregationNode::*)(class doris::vectorized::AggregationNode * const)) 0x5636308b69fa <doris::vectorized::AggregationNode::_close_without_key()>) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:111
#22 0x0000563630a4c6c6 in std::Bind_result<void, void (doris::vectorized::AggregationNode::(doris::vectorized::AggregationNode))()>::_call<void, , 0ul>(std::tuple<>&&, std::_Index_tuple<0ul>) (this=0x603017f3e9c0, __args=...)
at /var/local/ldb_toolchain/include/c++/11/functional:570
#23 0x0000563630a44903 in std::_Bind_result<void, void (doris::vectorized::AggregationNode::(doris::vectorized::AggregationNode))()>::operator()<>() (this=0x603017f3e9c0) at /var/local/ldb_toolchain/include/c++/11/functional:629
#24 0x0000563630a2f460 in std::_invoke_impl<void, std::_Bind_result<void, void (doris::vectorized::AggregationNode::(doris::vectorized::AggregationNode))()>&>(std::invoke_other, std::_Bind_result<void, void (doris::vectorized::AggregationNode::(doris::vectorized::AggregationNode))()>&) (_f=...) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:61
#25 0x0000563630a20eeb in std::__invoke_r<void, std::_Bind_result<void, void (doris::vectorized::AggregationNode::(doris::vectorized::AggregationNode))()>&>(std::_Bind_result<void, void (doris::vectorized::AggregationNode::(doris::vectorized::AggregationNode))()>&)
(__fn=...) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:111
#26 0x0000563630a1c681 in std::Function_handler<void (), std::_Bind_result<void, void (doris::vectorized::AggregationNode::(doris::vectorized::AggregationNode))()> >::_M_invoke(std::_Any_data const&) (_functor=...)
at /var/local/ldb_toolchain/include/c++/11/bits/std_function.h:291
#27 0x000056362e4a9be2 in std::function<void ()>::operator()() const (this=0x61e00107a9b8)
at /var/local/ldb_toolchain/include/c++/11/bits/std_function.h:560
#28 0x00005636308b210a in doris::vectorized::AggregationNode::close (this=0x61e00107a080, state=0x61c00084c080)
at /home/zcp/repo_center/doris_master/be/src/vec/exec/vaggregation_node.cpp:495
#29 0x000056362e5eadaa in doris::PlanFragmentExecutor::close (this=0x617000b589f0)
at /home/zcp/repo_center/doris_master/be/src/runtime/plan_fragment_executor.cpp:664
#30 0x000056362e59589f in doris::FragmentExecState::execute (this=0x617000b58980)
at /home/zcp/repo_center/doris_master/be/src/runtime/fragment_mgr.cpp:249
#31 0x000056362e59d9f1 in doris::FragmentMgr::_exec_actual(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>) (this=0x614000054c40, exec_state=..., cb=...)
at /home/zcp/repo_center/doris_master/be/src/runtime/fragment_mgr.cpp:499
#32 0x000056362e59fb4b in operator() (__closure=0x6080013c1220)
at /home/zcp/repo_center/doris_master/be/src/runtime/fragment_mgr.cpp:707
#33 0x000056362e5aa902 in std::_invoke_impl<void, doris::FragmentMgr::exec_plan_fragment(const doris::TExecPlanFragmentParams&, doris::FragmentMgr::FinishCallback)::<lambda()>&>(std::_invoke_other, struct
{...} &) (__f=...)
at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:61
#34 0x000056362e5aa4d8 in std::__invoke_r<void, doris::FragmentMgr::exec_plan_fragment(const doris::TExecPlanFragmentParams&, doris::FragmentMgr::FinishCallback)::<lambda()>&>(struct {...}
&) (__fn=...) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:111
#35 0x000056362e5a9e02 in std::Function_handler<void(), doris::FragmentMgr::exec_plan_fragment(const doris::TExecPlanFragmentParams&, doris::FragmentMgr::FinishCallback)::<lambda()> >::_M_invoke(const std::_Any_data &) (_functor=...)
at /var/local/ldb_toolchain/include/c++/11/bits/std_function.h:291
#36 0x000056362e4a9be2 in std::function<void ()>::operator()() const (this=0x6060042fab98)
at /var/local/ldb_toolchain/include/c++/11/bits/std_function.h:560
#37 0x000056362ecc5a60 in doris::FunctionRunnable::run (this=0x6060042fab90)
at /home/zcp/repo_center/doris_master/be/src/util/threadpool.cpp:45
#38 0x000056362ecc0dce in doris::ThreadPool::dispatch_thread (this=0x615000201c80)
at /home/zcp/repo_center/doris_master/be/src/util/threadpool.cpp:540
#39 0x000056362ece20c4 in std::__invoke_impl<void, void (doris::ThreadPool::&)(), doris::ThreadPool&> (
__f=@0x603017391220: (void (doris::ThreadPool::*)(class doris::ThreadPool * const)) 0x56362ecbfa9a <doris::ThreadPool::dispatch_thread()>, __t=@0x603017391230: 0x615000201c80) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:74
#40 0x000056362ece1963 in std::__invoke<void (doris::ThreadPool::&)(), doris::ThreadPool&> (
__fn=@0x603017391220: (void (doris::ThreadPool::*)(class doris::ThreadPool * const)) 0x56362ecbfa9a <doris::ThreadPool::dispatch_thread()>) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:96
#41 0x000056362ece0d02 in std::Bind<void (doris::ThreadPool::(doris::ThreadPool))()>::_call<void, , 0ul>(std::tuple<>&&, std::_Index_tuple<0ul>) (this=0x603017391220, __args=...) at /var/local/ldb_toolchain/include/c++/11/functional:420
#42 0x000056362ecdf813 in std::_Bind<void (doris::ThreadPool::(doris::ThreadPool))()>::operator()<, void>() (this=0x603017391220) at /var/local/ldb_toolchain/include/c++/11/functional:503
#43 0x000056362ecdc404 in std::_invoke_impl<void, std::_Bind<void (doris::ThreadPool::(doris::ThreadPool))()>&>(std::invoke_other, std::_Bind<void (doris::ThreadPool::(doris::ThreadPool))()>&) (_f=...) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:61
#44 0x000056362ecd9972 in std::_invoke_r<void, std::_Bind<void (doris::ThreadPool::(doris::ThreadPool))()>&>(std::_Bind<void (doris::ThreadPool::(doris::ThreadPool))()>&) (_fn=...) at /var/local/ldb_toolchain/include/c++/11/bits/invoke.h:111
#45 0x000056362ecd4c75 in std::Function_handler<void (), std::_Bind<void (doris::ThreadPool::(doris::ThreadPool))()> >::_M_invoke(std::_Any_data const&) (_functor=...) at /var/local/ldb_toolchain/include/c++/11/bits/std_function.h:291
#46 0x000056362e4a9be2 in std::function<void ()>::operator()() const (this=0x6110006e6bd8) at /var/local/ldb_toolchain/include/c++/11/bits/std_function.h:560
#47 0x000056362eca092a in doris::Thread::supervise_thread (arg=0x6110006e6bc0) at /home/zcp/repo_center/doris_master/be/src/util/thread.cpp:425
#48 0x00007f565a826609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#49 0x00007f565a960163 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95`

### What You Expected?

right 

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

Issue Number: close #13380 

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/vaggregation_node.cpp [hunk 1]
```diff
@@ -459,6 +459,7 @@ Status AggregationNode::open(RuntimeState* state) {
     // this could cause unable to get JVM
     if (_probe_expr_ctxs.empty()) {
         _create_agg_status(_agg_data.without_key);
+        _agg_data_created_without_key = true;
```
Comment (dataroaring) [inline be/src/vec/exec/vaggregation_node.cpp:462]: Could we use _probe_expr_ctxs.empty() as indicator, and do not introduce a new variable _agg_data_created_without_key.

**Review by mrhhsg — APPROVED** [review]

**Review by mrhhsg — APPROVED** [review]
LGTM

**Comment by hello-stephen** [comment]
https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221015033545_clickbench_pr_29540.html

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
1.1 not has this problem. Remove tags.

**Pull request merged**